no_std_test/src/main.rs

```diff
@@ -116,25 +111,32 @@ fn start(_argc: isize, _argv: *const *const u8) -> isize {
     let new_rec_sig = ecdsa::RecoverableSignature::from_compact(&data, rec_id).unwrap();
     assert_eq!(rec_sig, new_rec_sig);
 
-    let mut cbor_ser = [0u8; 100];
-    let writer = SliceWrite::new(&mut cbor_ser[..]);
-    let mut ser = Serializer::new(writer);
-    sig.serialize(&mut ser).unwrap();
-    let size = ser.into_inner().bytes_written();
-    let new_sig: ecdsa::Signature = de::from_mut_slice(&mut cbor_ser[..size]).unwrap();
-    assert_eq!(sig, new_sig);
-
     let _ = SharedSecret::new(&public_key, &secret_key);
     let _ = ecdh::shared_secret_point(&public_key, &secret_key);
 
+    #[cfg(feature = "use-serde")] {
```

Comment:
This style is inconsistent with the rest of the code and quite confusing.

Reply:
Do you mean inconsistent with the `rust-secp256k1` crate? This is a separate crate and the features are only used, as far as I can tell, to run this crate and selectively run parts of the test (`start` function).

FTR I'm not sure why this is a separate crate, I'm not super deep in the no-std stuff, @elichai do you remember off the top of your head why we had to put it in a separate crate and not in the `tests` directory (i.e., standard integration tests directory)?

Reply:
The goal was to simulate macro usage from `rust-bitcoin`, which has less visibility of symbols than anything within the crate (including, I think (?) stuff in the `tests/` directory).

I believe this also predates the `tests` directory.